R/ctwas_wrapper.R

```diff
@@ -93,79 +94,12 @@ generate_twas_db <- function(weight_db_file, contexts = NULL, variable_name_obj
-              # Filter out any indices that have already been selected
-              available_indices <- setdiff(cs, selected_indices)
-              if (length(available_indices) > 0) {
-                # Get the index with the highest PIP score
-                top_index <- available_indices[which.max(set_cs_list$pip[available_indices])]
-                selected_indices <- c(selected_indices, top_index)
-              }
-            }
-          }
-        }
-        return(selected_indices)
-      }
-      # check if the context has top_loci table
-      if ("top_loci" %in% names(twas_data_combined$susie_results[[context]])) {
-        top_loci <- twas_data_combined$susie_results[[context]][["top_loci"]]
-        if (length(top_loci[, "variant_id"]) <= max_var_selection) {
-          variant_selected <- top_loci$variant_id
-        } else {
-          # if top_loci variants more than max_var_selection, we loop through CS set to select top pip variants from each CS set
-
-          if (!is.null(susie_result_trimmed$sets$cs)) {
-            # check total variant number in the CS sets
-            totl_cs_indice <- do.call(sum, lapply(names(susie_result_trimmed$sets$cs), function(L) {
-              length(susie_result_trimmed$sets$cs[[L]])
-            }))
-            if (max_var_selection <= totl_cs_indice) {
-              selec_idx <- select_cs_var(susie_result_trimmed, max_var_selection)
-              variant_selected <- variant_names[selec_idx]
-            } else {
-              # when top loci has larger number of variants than max_var_selection, but CS sets has less number than max_var_selection
-              selec_idx <- unlist(susie_result_trimmed$sets$cs)
-              variant_selected <- variant_names[selec_idx]
-            }
-          } else {
-            top_idx <- order(susie_result_trimmed$pip, decreasing = TRUE)[1:max_var_selection]
-            variant_selected <- variant_names[top_idx]
-          }
-        }
-      } else {
-        # if the context did not come with the top_loci table, we select top pip variants from [[context]]$preset_variants_result$susie_result_trimmed$pip
-        print(paste0(context, " do not have top_loci table, select top pip variants. "))
-        top_idx <- order(susie_result_trimmed$pip, decreasing = TRUE)[1:max_var_selection]
-        variant_selected <- variant_names[top_idx]
-      }
-      return(variant_selected)
-    })
-    names(var_selection_list) <- contexts
-    return(var_selection_list)
-  }
-
   ## load twas weights
-  twas_data_combined <- load_twas_weights(weight_db_file, conditions = contexts, variable_name_obj = variable_name_obj, 
+  twas_data_combined <- load_twas_weights(weight_db_file, conditions = contexts, variable_name_obj = variable_name_obj, susie_obj=susie_obj,
                                   twas_weights_table = twas_weights_table)
+  if(!"weights" %in% names(twas_data_combined)) return(NULL)
```

Comment:
how is it possible that "weights" is not in the twas_data object? IF it is not expect should'nt you throw an error here?

Reply:
that is indeed should be throw out as error, I will have that corrected

Reply:
This is fixed now